sdk/src/record-scanner.ts

```diff
@@ -0,0 +1,316 @@
+import { Account } from "./account";
+import { EncryptedRecord } from "./models/record-provider/encryptedRecord";
+import { OwnedFilter } from "./models/record-scanner/ownedFilter";
+import { OwnedRecord } from "./models/record-provider/ownedRecord";
+import { RecordProvider } from "./record-provider";
+import { RecordPlaintext } from "./wasm";
+import { RecordSearchParams } from "./models/record-provider/recordSearchParams";
+import { RecordsFilter } from "./models/record-scanner/recordsFilter";
+import { RecordsResponseFilter } from "./models/record-provider/recordsResponseFilter";
+import { RegistrationRequest } from "./models/record-scanner/registrationRequest";
+
+/**
+ * RecordScanner is a RecordProvider implementation that uses the record scanner service to find records.
+ * 
+ * @example
+ * const account = new Account({ privateKey: 'APrivateKey1...' });
+ * 
+ * const recordScanner = new RecordScanner("https://record-scanner.aleo.org");
+ * recordScanner.setAccount(account);
+ * await recordScanner.register(0);
+ * 
+ * const filter = {
+ *     start: 0,
+ *     end: 100,
+ *     program: "credits.aleo",
+ *     record: "credits",
+ * };
+ * 
+ * const responseFilter = {
+ *     program: true,
+ *     record: true,
+ *     function: true,
+ *     transition: true,
+ *     blockHeight: true,
+ *     transactionId: true,
+ * };
+ * 
+ * const records = await recordScanner.findRecords({ filter, responseFilter });
+ */
+class RecordScanner implements RecordProvider {
+    account?: Account;
+    readonly url: string;
+    private uuid?: string;
+
+    constructor(url: string, account?: Account) {
+        this.account = account;
+        this.url = url;
+    }
+    
+    /**
+     * Set the account to use for the record scanner.
+     * 
+     * @param {Account} account The account to use for the record scanner.
+     */
+    async setAccount(account: Account): Promise<void> {
+        this.uuid = undefined;
+        this.account = account;
+    }
+
+    /**
+     * Register the account with the record scanner service.
+     * 
+     * @param {number} startBlock The block height to start scanning from.
+     * @returns {Promise<void>} A promise that resolves when the account is registered.
+     */
+    async register(startBlock: number): Promise<void> {
```

Comment:
This should return a `RegistrationResponse` type

```js
interface RegistrationResponse {
   uuid: string,
   job_id?: string,
   status?: string 
}
```

```suggestion
    async register(startBlock: number): Promise<void> {
```